src/main/java/science/icebreaker/network/NetworkService.java

```diff
@@ -98,6 +102,11 @@ public int countNodes() {
     }
 
 
+    public List<Paper> getPapers(List<Integer> ids) {
+        return paperRepository.findAllByIds(ids);
```

Comment:
So you always return all papers to the frontend? How does that scale with the amount of papers in the database.

Reply:
It's only returning those that the frontend wants and the frontend is only requesting those corresponding to an edge that was just clicked.